onnxruntime/core/providers/js/operators/reduce.cc

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "reduce.h"
+
+namespace onnxruntime {
+namespace js {
+
+#define REGISTER_REDUCE_ELEMENTWISE_VERSIONSED_KERNEL(ReduceOp, sinceVersion, endVersion) \
+  ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_EX(                                                \
+      ReduceOp,                                                                           \
+      kOnnxDomain,                                                                        \
+      sinceVersion, endVersion,                                                           \
+      float,                                                                              \
+      kJsExecutionProvider,                                                               \
+      (*KernelDefBuilder::Create())                                                       \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),                     \
+      ReduceOp<float>);                                                                   \
+                                                                                          \
+  ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_EX(                                                \
```

Comment:
[nit] if possible, use non-typed declaration of kernel classes. we don't need a template parameter for datatype because our C++ code does not do the actual computation. this may slightly benefits the binary size.